pkgs/build-support/docker/default.nix

```diff
@@ -35,6 +35,9 @@
 , writePython3
 }:
 
+# Prevent infinite recursion error to happen with pkgs.pkgsCross.[ARCH]-darwin
+assert go.GOOS == "darwin" -> throw "can't build container image for `darwin` with `dockerTools`";
```

Comment:
I also get this error even with the `hello` package, so it's not specific to dockerTools.

```suggestion
```